src/app/fdctl/config.c

```diff
@@ -759,7 +759,12 @@ config_parse( int *    pargc,
   } else {
     if( FD_UNLIKELY( !if_nametoindex( result.tiles.net.interface ) ) )
       FD_LOG_ERR(( "configuration specifies network interface `%s` which does not exist", result.tiles.net.interface ));
-    result.tiles.net.ip_addr = listen_address( result.tiles.net.interface );
+      
+    uint iface_ip = listen_address( result.tiles.net.interface );
+    if ( FD_UNLIKELY( !fd_ip4_addr_is_public ( iface_ip ) ) )
+      FD_LOG_ERR(( "configuration specifies network interface `%s` which does not have a public IP address", result.tiles.net.interface ));
```

Comment:
The comment could be a little wrong, since the configuration can have an empty interface which will get autodetected above. How about

```suggestion
      FD_LOG_ERR(( "Trying to use network interface `%s` for listening to incoming transactions, but it has IPv4 address " FD_IP4_ADDR_FMT " which is part of a private network and will not be routable for other Solana network nodes.", result.tiles.net.interface, FD_IP4_ADDR_FMT_ARGS( iface_ip ) ))
```